src/adapters/base.js

```diff
@@ -16,7 +16,25 @@ export default class BaseAdapter {
   }
 
   /**
-   * Adds a record to the IndexedDB database.
+   * Opens a connection o the database.
+   *
+   * @return {Promise}
+   */
+  open() {
+    throw new Error("Not Implemented.");
```

Comment:
I would argue that base could do what you've currently done in LocalStorage - that way, other adapters that don't need to explicitly manage connections don't have to worry about implementing them. If we throw when these are unimplemented, this is a breaking change.